docs/001-mana_proposal.md

```diff
@@ -0,0 +1,539 @@
+    of the two). Trigger `ManaUpdated` event.
+ - `GetConsensusMana(nodeID) mana`: access `Base Mana Vector` of `Consensus Mana`, update its values with respect to time,
+    and returns the amount of `Consensus Mana` (either `Effective Base Mana 1`, `Effective Base Mana 2`, or some combination
+    of the two). Trigger `ManaUpdated` event.
+ - `GetNeighborsMana(type)`: returns the `type` mana of the nodes neighbors
+ - `GetAllManaVectors()` Obtaining the full mana maps for comparison with the perception of other nodes.
+ - `GetWeightedRandomNodes(n)`: returns a weighted random selection of `n` nodes. `Consensus Mana` is used for the weights.
+ - Obtaining a list of currently known peers + their mana, sorted. Useful for knowing which high mana nodes are online.
+ - `OverrideMana(nodeID, baseManaVector)`: Sets the nodes mana to a specific value. Can be useful for debugging, setting faucet mana, initialization, etc.. Triggers `ManaUpdated`
+
+Such utility functions could be used for example to visualize mana distribution in node dashboard, or send neighbor
+mana data to the analysis server for further processing.
+
+#### Booking Mana
+Mana is booked when a transaction is confirmed.
+
+ ```go
+on TransactionConfirmed (tx):
+  bookAccessMana()
+  bookConsensusMana()
+```
+
+#### Synchronization and Mana Calculation
+
+The mana plugin is responsible to determine when to start calculating mana locally.
+Since mana state is an extension to ledger state, it can only depict realistic mana values once the node is in sync.
+During syncing, ledger state is constructed from messages coming from neighbors as described further above.
+
+In this first iteration, mana plugin relies on `TransactionConfirmed` event of the value transfers plugin, and has no
+explicit rules on when to start and stop mana calculation.
+
+In future, initial mana state (together with the initial ledger state) will be derived from a snapshot file.
+
+### Mana Toolkit
+In this section, all tools and utility functions for mana will be outlined.
+
+#### Mana Related API endpoints
+ - `/info`: Add own mana in node info response.
+ - `value/allowedManaPledge`: Endpoint that clients can query to determine which nodeIDs are allowed as part of
+   `accessMana` and `consensusMana` fields in a transaction.
+ - `value/sendTransactionByJson`: Add `accessMana`, `consensusMana` and `timestamp` fields to the JSON request.
+
+Add a new `mana` endpoint route:
+ - `/mana`: Return access and consensus mana of the node.
+ - `/mana/all`: Return whole mana map (mana perception of the node).
+ - `/mana/access/nhighest`: Return `n` highest access mana holder `nodeIDs` and their access mana values.
+ - `/mana/consensus/nhighest`: Return `n` highest consensus mana holder `nodeIDs` and their consensus mana values.
+ - `/mana/rank`: Return the top percentile the node belongs to relative to the network. For example, if there are 100 nodes in the
```

Comment:
I'm afraid this naming could create confusion with `ManaRank` used within the autopeering

Reply:
let's call it `mana/percentile` then.